ch2driver/pytpcc/runtime/loader.py

```diff
@@ -520,7 +523,9 @@ def generateCH2Order(self, o_w_id, o_d_id, o_id, o_c_id, o_ol_cnt, orderTime, ne
         o_entry_d = orderTime
         o_carrier_id = constants.NULL_CARRIER_ID if newOrder else rand.number(constants.MIN_CARRIER_ID, constants.MAX_CARRIER_ID)
         o_all_local = constants.INITIAL_ALL_LOCAL
-        return [ o_id, o_c_id, o_d_id, o_w_id, o_entry_d, o_carrier_id, o_ol_cnt, o_all_local ]
+        o_tuple = [ o_id, o_c_id, o_d_id, o_w_id, o_entry_d, o_carrier_id, o_ol_cnt, o_all_local ]
+        o_tuple.append(self.generateExtraFields(self.ordersExtraFields))
+        return o_tuple
```

Comment:
Did think about it, but decided to keep it separate just in case we decide to change orders with more nesting or other stuff in the future. I did think of simplifying the generation of all the other collections too, but with the nesting differences (in warehouse, supplier, etc.) it would not be easy. Therefore, I decided to keep the generation of orders also separate for consistency and for future proofing. Thoughts?

Reply:
I think that's fair - CH2 is pretty "locked in" now, while CH2++ is much more likely to be modified, and the current approach does make it easier to update CH2++ completely independently of CH2 (without accidentally introducing a bug into CH2).